File: geoDistLRUCache/Node.py
from typing import Dict, Optional


class Node:
    """
    Linked List Node. Contains key-value pair and links to neighbor elements.
    """

    def __init__(self, key: int, value: int, expiration: int, prev=None, next=None):
        self.expiration = expiration
        self.key: int = key
        self.value: int = value
        self.prev: Optional[Node] = prev
        self.next: Optional[Node] = next

File: distributedCache/Node.cpp
#include <memory>

class Node {
public:
    int expiration;
    int key;
    int value;
    std::shared_ptr<Node> prev;
    std::shared_ptr<Node> next;

    Node(int key, int value, int expiration, std::shared_ptr<Node> prev = nullptr, std::shared_ptr<Node> next = nullptr) 
        : key(key), value(value), expiration(expiration), prev(prev), next(next) {}
};
